vms/avm/index_test.go

```diff
@@ -5,349 +5,177 @@ package avm
-			inputUTXOs = append(inputUTXOs, utxo)
-		}
+		issueAndAccept(require, env.vm, env.issuer, tx)
 
-		// index the transaction
-		require.NoError(vm.addressTxsIndexer.Accept(uniqueParsedTX.ID(), inputUTXOs, uniqueParsedTX.UTXOs()))
+		addressTxMap[addr] = tx
 	}
 
 	// ensure length is same as keys length
 	require.Len(addressTxMap, len(keys))
 
 	// for each *UniqueTx check its indexed at right index for the right address
-	for key, tx := range addressTxMap {
-		assertIndexedTX(t, vm.db, uint64(0), key, txAssetID.ID, tx.ID())
-		assertLatestIdx(t, vm.db, key, txAssetID.ID, 1)
+	for addr, tx := range addressTxMap {
+		assertIndexedTX(t, env.vm.db, 0, addr, txAssetID.ID, tx.ID())
+		assertLatestIdx(t, env.vm.db, addr, txAssetID.ID, 1)
 	}
 }
 
 func TestIndexTransaction_MultipleAddresses(t *testing.T) {
 	require := require.New(t)
 
-	genesisBytes := BuildGenesisTest(t)
-	issuer := make(chan common.Message, 1)
-	baseDBManager := manager.NewMemDB(version.Semantic1_0_0)
-	ctx := NewContext(t)
-	genesisTx := GetAVAXTxFromGenesisTest(genesisBytes, t)
-
-	avaxID := genesisTx.ID()
-	vm := setupTestVM(t, ctx, baseDBManager, genesisBytes, issuer, indexEnabledAvmConfig)
+	env := setup(t, &envConfig{
+		vmStaticConfig: &config.Config{},
+	})
 	defer func() {
-		require.NoError(vm.Shutdown(context.Background()))
-		ctx.Lock.Unlock()
+		require.NoError(env.vm.Shutdown(context.Background()))
+		env.vm.ctx.Lock.Unlock()
 	}()
 
-	txAssetID := avax.Asset{ID: avaxID}
 	addrs := make([]ids.ShortID, len(keys))
-	for _, key := range keys {
-		addrs = append(addrs, key.PublicKey().Address())
+	for i, key := range keys {
+		addrs[i] = key.PublicKey().Address()
 	}
+	utils.Sort(addrs)
 
-	ctx.Lock.Lock()
+	txAssetID := avax.Asset{ID: env.genesisTx.ID()}
 
 	key := keys[0]
 	addr := key.PublicKey().Address()
-	// create utxoID and assetIDs
+
+	// make utxo
 	utxoID := avax.UTXOID{
 		TxID: ids.GenerateTestID(),
 	}
+	utxo := buildUTXO(utxoID, txAssetID, addr)
+	env.vm.state.AddUTXO(utxo)
 
-	// build the transaction
+	// make transaction
 	tx := buildTX(utxoID, txAssetID, addrs...)
+	require.NoError(tx.SignSECP256K1Fx(env.vm.parser.Codec(), [][]*secp256k1.PrivateKey{{key}}))
 
-	// sign the transaction
-	require.NoError(signTX(vm.parser.Codec(), tx, key))
-
-	// Provide the platform UTXO
-	utxo := buildPlatformUTXO(utxoID, txAssetID, addr)
-
-	// save utxo to state
-	vm.state.AddUTXO(utxo)
-
-	var inputUTXOs []*avax.UTXO //nolint:prealloc
-	for _, utxoID := range tx.Unsigned.InputUTXOs() {
-		utxo, err := vm.state.GetUTXO(utxoID.InputID())
-		require.NoError(err)
-
-		inputUTXOs = append(inputUTXOs, utxo)
-	}
+	// issue transaction
+	issueAndAccept(require, env.vm, env.issuer, tx)
 
-	// index the transaction
-	require.NoError(vm.addressTxsIndexer.Accept(tx.ID(), inputUTXOs, tx.UTXOs()))
-
-	assertIndexedTX(t, vm.db, uint64(0), addr, txAssetID.ID, tx.ID())
-	assertLatestIdx(t, vm.db, addr, txAssetID.ID, 1)
-}
-
-func TestIndexTransaction_UnorderedWrites(t *testing.T) {
```

Comment:
This test was an **_exact_** copy of `TestIndexTransaction_MultipleTransactions` (except for a comment and whitespace)